src/rust/engine/src/nodes.rs

```diff
@@ -1118,7 +1118,15 @@ impl Task {
                 ))
               }
             })?;
-          select.run_node(context).await
+          let res = select.run_node(context).await;
+          if get.safe {
+            match res {
+              Err(Failure::Throw { val, .. }) => Ok(val),
+              _ => res,
+            }
+          } else {
+            res
+          }
```

Comment:
Generators have a `generator.throw` method which should probably be used rather than sending an `Exception` value via `generator.send`: otherwise `Get`s will not be able to request values which subclass `Exception` _and_ act as error boundaries (niche, I know).

Reply:
@stuhood Well, that's neat, and would be cool to support (however niche), but not even the prior art you linked to supported this (although using the `generator.throw` idiom) as it was switching to it based on `if isinstance(arg, Exception)`:

https://github.com/pantsbuild/pants/pull/10954/files#diff-156757f090beeedd1a880772f81cae2da74e4275b55d392b2cc06a44543c3ee9R106


I'm working on reviving #10954 ...

Reply:
Also, careful with Exception v BaseException. I never know when it's appropriate to use Base instead of Exception

Reply:
> Also, careful with Exception v BaseException. I never know when it's appropriate to use Base instead of Exception.
> 
> Edit: ah I think it's exceptions that terminate don't derive from Exception ?

Yea, I think I recall having read that `Exception` is everything except `SystemExit` and `KeyboardInterrupt` pretty much..